recipes/llhttp/all/conanfile.py

```diff
@@ -0,0 +1,90 @@
+from conans import ConanFile, tools, CMake
+from conan.tools.files import patch
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class LlhttpParserConan(ConanFile):
+    name = "llhttp"
+    description = "http request/response parser for c "
+    topics = ("http", "parser")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nodejs/llhttp"
+    license = ("MIT",)
+    generators = ("cmake",)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    exports_sources = "CMakeLists.txt", "patches/*"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            destination=self._source_subfolder,
+            strip_root=True,
+        )
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(
+            "LICENSE-MIT",
+            src=os.path.join(self.source_folder, self._source_subfolder),
+            dst="licenses",
+        )
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "llhttp")
+        self.cpp_info.set_property("cmake_target_name", "llhttp::llhttp")
+        self.cpp_info.set_property("pkg_config_name", "libllhttp")
+        self.settings.compiler == "Visual Studio"
+        postfix = (
+            "d"
+            if self.settings.compiler == "Visual Studio"
+            and self.settings.build_type == "Debug"
+            else ""
```

Comment:
seems like postfix doesn't (always?) match:
```
llhttp/6.0.6 package(): Packaged 1 '.lib' file: llhttp.lib
```
(from debug build log)

Reply:
- I think there is no debug postfix I think because they declare add_library() before CMAKE_DEBUG_POSTFIX: https://github.com/nodejs/llhttp/blob/v6.0.6/CMakeLists.txt#L32-L34
- Shared build for msvc doesn't export symbols, you can add CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON in CMake wrapper.